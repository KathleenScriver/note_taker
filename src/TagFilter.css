.tag-filter {
  display: flex;
  justify-content: center;
}

.tag-filter button {
  margin: 10px;
  background-color: #9AD4D6;
  border-color: #538083;
  border-width: 1px;
  border-radius: 10px;
  padding: 5px;
  color: white;
  width: 70px;
  text-align: center;
  font-size: 14px;
}
