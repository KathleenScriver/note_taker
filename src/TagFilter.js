import React, { Component } from 'react';
import './TagFilter.css';


export default class TagFilter extends Component {
  render() {
    return (
      <div className="tag-filter">
        <button onClick={this.filterNotes} value="work">Work</button>
        <button onClick={this.filterNotes} value="hobby">Hobby</button>
        <button onClick={this.filterNotes} value="personal">Personal</button>
      </div>
    )
  }
}
