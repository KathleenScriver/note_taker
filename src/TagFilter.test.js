import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import TagFilter from './TagFilter';

describe(TagFilter, () => {
  const mockFilterNotes = jest.fn();
  const component = shallow(
    <TagFilter filterNotes={mockFilterNotes}/>
  );

  it('renders and matches our snapshot', () => {
    const component = renderer.create(
      <TagFilter />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('should have filter buttons', () => {
    expect(component.find('button')).toHaveLength(4);
  })

  it('calls passed in filterNotes function when clicked', () => {
    component.find('button.work').simulate('click');
    expect(mockFilterNotes).toBeCalled();
  })
})
