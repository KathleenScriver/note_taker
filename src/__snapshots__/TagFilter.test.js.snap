// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagFilter renders and matches our snapshot 1`] = `
<div
  className="tag-filter"
>
  <button
    className="work"
    onClick={[Function]}
    value="work"
  >
    Work
  </button>
  <button
    className="hobby"
    onClick={[Function]}
    value="hobby"
  >
    Hobby
  </button>
  <button
    className="personal"
    onClick={[Function]}
    value="personal"
  >
    Personal
  </button>
  <button
    className="all"
    onClick={[Function]}
    value="all"
  >
    All
  </button>
</div>
`;
